lib/permutations.rb

```diff
@@ -1,4 +1,40 @@
 
```

Comment:
For lines 8 & 16, see comment in palindrome_permutation.rb about hashtable vs array lookup.